datafusion/expr/src/udf.rs

```diff
@@ -646,6 +653,11 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
         Ok(SortProperties::Unordered)
```

Comment:
Instead of directly returning Unordered, lets add an if check:
```rust
if self.output_preserves_lex_ordering() {
    // now look the input orders, and if they are all going to the same direction (all asc or dsc), the output order will be that direction 
    ...
    } else {
        Unordered
    }
   ```

Reply:
Resolved in 8949021.